### PR TITLE
Add / Document secureCodeBox Operator Prometheus Metrics

### LINKED ISSUE #2580: Prometheus Metrics for the secureCodeBox Operator
### Is your feature request related to a problem?

It can be hard to monitor larger secureCodeBox setups with hundreds / thousands of scans a day and get a good idea of which scans are succeeding / failing.

### Describe the solution you'd like

We are already exporting some prometheus metrics which are automatically provided by kubebuilder.
It should be documented on how to access them.

We should also start tracking our own metrics which might be relevant to keep an eye on.

### Describe alternatives you've considered

Log monitoring. Less efficient and harder to setup and configure properly.

### PR DESCRIPTION
**Pull request by J12934**
## Description

- Add integration / helm value to let prometheus scrape secureCodeBox operator metrics
- Document operator metrics / where to configure them
- Add three custom metrics to begin with to track how many scans per scan_type were created and succeeded / failed
  - Refactored Scan status for this to a enum to better allow to track all "state" transitions

Closes #2580 

### Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] Make sure that all your commits are signed-off and that you are added to the [Contributors](https://github.com/secureCodeBox/secureCodeBox/blob/main/CONTRIBUTORS.md) file.
* [ ] Make sure that all CI finish successfully.
* [ ] Optional (but appreciated): Make sure that all commits are [Verified](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).

### CODE REVIEW

File: documentation/docs/getting-started/installation.md [hunk 1]
```diff
@@ -91,6 +91,21 @@ s3:
   keySecret: gcs-bucket-credentials
 ```
 
+### Prometheus Metrics
+
+If you want to monitor the scans managed by your secureCodeBox Operator you can use the [Prometheus](https://prometheus.io/) metrics that the operator provides.
```
Comment (Weltraumschaf) [inline documentation/docs/getting-started/installation.md:96]:
```suggestion
If you want to monitor the scans managed by your _secureCodeBox Operator_ you can use the [Prometheus](https://prometheus.io/) metrics that the operator provides.
```

File: documentation/docs/getting-started/installation.md [hunk 2]
```diff
@@ -91,6 +91,21 @@ s3:
...
+To use them you need to configure your prometheus instance to scrape these metrics from port 8080 of the operator pod.
+
+If you are using the prometheus operator you can enable metric collection by enabling the service monitoring included in the operators helm chart.
```
Comment (Weltraumschaf) [inline documentation/docs/getting-started/installation.md:99]:
```suggestion
If you are using the Prometheus operator you can enable metric collection by enabling the service monitoring included in the operators helm chart.
```

File: documentation/docs/getting-started/installation.md [hunk 3]
```diff
@@ -91,6 +91,21 @@ s3:
...
+To do that you need to set the `metrics.serviceMonitor.enabled` flag to `true`.
+
+```bash
+helm --namespace securecodebox-system upgrade --install --create-namespace securecodebox-operator oci://ghcr.io/securecodebox/helm/operator --set="metrics.serviceMonitor.enabled=true"
+```
+
+Custom metrics from the secureCodeBox operator are prefixed with: `securecodebox_`.
```
Comment (Weltraumschaf) [inline documentation/docs/getting-started/installation.md:106]:
```suggestion
Custom metrics from the _secureCodeBox Operator_ are prefixed with: `securecodebox_`.
```

File: documentation/docs/getting-started/installation.md [hunk 4]
```diff
@@ -91,6 +91,21 @@ s3:
...
+You can list them, together with an explanation in your prometheus / grafana explore view.
```
Comment (Weltraumschaf) [inline documentation/docs/getting-started/installation.md:107]:
```suggestion
You can list them, together with an explanation in your Prometheus/Grafana explore view.
```

File: operator/controllers/execution/scans/scan_controller.go [hunk 5]
```diff
@@ -43,6 +44,22 @@ var (
...
+	ScanStatusParsing                    ScanStatus = "Parsing"
+	ScanStatusParseCompleted             ScanStatus = "ParseCompleted"
+	ScanStatusHookProcessing             ScanStatus = "HookProcessing"
+	ScanStatusReadAndWriteHookProcessing ScanStatus = "ReadAndWriteHookProcessing"
+	ScanStatusReadAndWriteHookCompleted  ScanStatus = "ReadAndWriteHookCompleted"
+	ScanStatusReadOnlyHookProcessing     ScanStatus = "ReadOnlyHookProcessing"
+	ScanStatusErrored                    ScanStatus = "Errored"
+	ScanStatusDone                       ScanStatus = "Done"
+)
+
```
Comment (Weltraumschaf) [inline operator/controllers/execution/scans/scan_controller.go:62]: Why copy pasted from operator/apis/execution/v1/scan_types.go?

**Review by Weltraumschaf — COMMENTED** [review]
See my suggestions/comments.

File: operator/controllers/execution/scans/scan_controller.go (hunk 5 above)
Comment (J12934) [reply operator/controllers/execution/scans/scan_controller.go]: Good point, had it there initially, then decided to move it.
Forgot to remove it here. Thanks for catching it 🙌
It's removed now

**Review by Weltraumschaf — APPROVED** [review]

**Pull request merged**